### PR TITLE
FileStorage

### PR DESCRIPTION
**Pull request by panarch**
FileStorage is a custom storage implementation that utilizes the filesystem.

For each table name, the schema information is saved in a `.sql` format using a `CREATE TABLE` query.

The data is stored by creating a directory with the same name as the table and serializing the data using [RON](https://github.com/ron-rs/ron) format in the subdirectory.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -67,6 +67,7 @@ jobs:
           cd json-storage && cargo test --verbose && cd ..
           cd csv-storage && cargo test --verbose && cd ..
           cd parquet-storage && cargo test --verbose && cd ..
+          cs file-storage && cargo test --verbose && cd ..
```
Comment (devgony) [inline .github/workflows/rust.yml:70]: is `cs` typro?

Comment (panarch) [reply .github/workflows/rust.yml:70]: ooooh, thanks a lot!

File: storages/file-storage/src/store.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+#[async_trait(?Send)]
+impl Store for FileStorage {
+    async fn fetch_all_schemas(&self) -> Result<Vec<Schema>> {
+        let mut schemas = fs::read_dir(&self.path)
+            .map_storage_err()?
+            .map(|dir_entry| {
+                let dir_entry = dir_entry.map_storage_err()?;
+                let file_type = dir_entry.file_type().map_storage_err()?;
+                if file_type.is_dir() {
```
Comment (devgony) [inline storages/file-storage/src/store.rs:21]:
```suggestion
                let path = dir_entry.path();
                let extension = path.extension().and_then(OsStr::to_str);
                if file_type.is_dir() || extension != Some("sql") {
```

Would it be better to validate if it's *.sql as well?

Comment (devgony) [reply storages/file-storage/src/store.rs:21]: Or.. maybe we can reuse fetch_schema

File: storages/file-storage/src/store.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        let mut entries = fs::read_dir(path)
+            .map_storage_err()?
+            .map(|dir_entry| {
+                let dir_entry = dir_entry.map_storage_err()?;
+                let file_type = dir_entry.file_type().map_storage_err()?;
+                if file_type.is_dir() {
+                    return Ok(None);
+                }
+
+                Ok(Some(dir_entry.path()))
```
Comment (devgony) [inline storages/file-storage/src/store.rs:71]:
```suggestion
                let path = dir_entry.path();
                let extension = path.extension().and_then(OsStr::to_str);
                if file_type.is_dir() || extension != Some("ron") {
                    return Ok(None);
                }

                Ok(Some(path))
```

Similar pattern exists in scan_data !

**Review by devgony — APPROVED** [review]

**Pull request merged**